adapter/oam.go

```diff
@@ -163,6 +165,92 @@ func (or *OAMRegistrant) Register() error {
 	return nil
 }
 
+//StaticCompConfig is used to configure CreateComponents
+type StaticCompConfig struct {
+	URL     string           //URL
+	Method  string           //Use the constants exported by package. Manifests or Helm
+	Path    string           //Where to store the directory.(Each directory will have an array of definitions and schemas)
+	DirName string           //The directory's name. By convention, it should be the version name
+	Config  manifests.Config //Filters required to create definition and schema
+	Force   bool             //When set to true, if the file with same name already exists, they will be overridden
+}
+
+//CreateComponents generates components for a given configuration and stores them.
+func CreateComponents(scfg StaticCompConfig) error {
+	dir := filepath.Join(scfg.Path, scfg.DirName)
+	_, err := os.Stat(dir)
+	if err != nil && !os.IsNotExist(err) {
+		return ErrCreatingComponents(err)
+	}
+	if err != nil && os.IsNotExist(err) {
+		err = os.Mkdir(dir, 0777)
+		if err != nil {
+			return ErrCreatingComponents(err)
+		}
+	}
+	var comp *manifests.Component
+	switch scfg.Method {
+	case Manifests:
+		comp, err = manifests.GetFromManifest(scfg.URL, manifests.SERVICE_MESH, scfg.Config)
+	case HelmCHARTS:
+		comp, err = manifests.GetFromHelm(scfg.URL, manifests.SERVICE_MESH, scfg.Config)
+	default:
+		return ErrCreatingComponents(errors.New("invalid generation method. Must be either Manifests or HelmCharts"))
+	}
+	if err != nil {
+		return ErrCreatingComponents(err)
+	}
+	if comp == nil {
+		return ErrCreatingComponents(errors.New("nil components"))
+	}
+	for i, def := range comp.Definitions {
+		schema := comp.Schemas[i]
+		name := getNameFromWorkloadDefinition([]byte(def))
+		defFileName := name + "_definition.json"
+		schemaFileName := name + ".meshery.layer5io.schema.json"
+		err := writeToFile(filepath.Join(dir, defFileName), []byte(def), scfg.Force)
+		if err != nil {
+			return ErrCreatingComponents(err)
+		}
+		err = writeToFile(filepath.Join(dir, schemaFileName), []byte(schema), scfg.Force)
+		if err != nil {
+			return ErrCreatingComponents(err)
+		}
+	}
+	return nil
+}
+
+//create a file with this filename and stuff the string
+func writeToFile(path string, data []byte, force bool) error {
+	_, err := os.Stat(path)
+	if err != nil && !os.IsNotExist(err) { //There some other error than non existence of file
+		return err
+	}
+
+	if err == nil { //file already exists
+		if !force { // Dont override existing file, skip it
+			fmt.Println("File already exists,skipping...")
+			return nil
+		}
+		err := os.Remove(path) //Remove the existing file, before overriding it
+		if err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(path, data, 0777)
+}
+
+//getNameFromWorkloadDefinition takes out name from workload definition
+func getNameFromWorkloadDefinition(definition []byte) string {
```

Comment:
Is this name being run through string manipulation for display formatting?

Reply:
No it's not.

Reply:
Maybe this isn't an appropriate location for such manipulation. Where is the central set of string manipulation functions? Are they referenced by this static component generation functions?

Reply:
I am not manipulating anything here. I am just extracting the names to set the names of the files.

Reply:
Where is the single location in which string manipulation logic is kept?

Reply:
We can follow up this conversation post-merge. 👍

Reply:
Merged. Address the question above when you have time.

Reply:
String manipulation is also present in meshkit function